pkg/daemon/sync.go

```diff
@@ -41,67 +44,103 @@ type changeSet struct {
 }
 
 // Sync starts the synchronization of the cluster with git.
-func (d *Daemon) Sync(ctx context.Context, started time.Time, newRevision string, ratchet revisionRatchet) error {
-	// Make a read-only clone used for this sync
+func (d *Daemon) Sync(ctx context.Context, started time.Time, newRevision string, ratchet ratchet) error {
+	currentRevision, err := ratchet.CurrentRevision(ctx)
+	if err != nil {
+		return err
+	}
+
 	ctxt, cancel := context.WithTimeout(ctx, d.GitTimeout)
```

Comment:
Creating this context here and using it in (potentially) two places below is odd. The purpose of making a new context (rather than using the one passed as an argument) is to give each git operation a timeout; so, _each_ git operation should get a context with a deadline, just before it's attempted.

Reply:
I understood and fixed. Thank you.